packages/react-router/src/link.tsx

```diff
@@ -242,25 +248,22 @@ type PostProcessParams<
 > = TParamVariant extends 'SEARCH' ? ExcludeRootSearchSchema<T> : T
 
 type ResolveFromParams<
-  TRouteTree extends AnyRoute,
+  TRouter extends AnyRouter,
   TParamVariant extends ParamVariant,
   TFrom,
-  TFromRouteType extends
-    | 'allParams'
-    | 'fullSearchSchema' = TParamVariant extends 'PATH'
-    ? 'allParams'
-    : 'fullSearchSchema',
 > = PostProcessParams<
-  RouteByPath<TRouteTree, TFrom>['types'][TFromRouteType],
+  RouteByPath<TRouter['routeTree'], TFrom>['types'][TParamVariant extends 'PATH'
```

Comment:
is this a typescript performance optimization (moving out of the type params)?

Reply:
No. I thought about it quite a bit. I now need access to the options on the router. Which would be fine with `RegisteredRouter` if it weren't for the type tests. Module augmentation is awkward with type tests. It's easier to mock it out if the whole router is a type parameter